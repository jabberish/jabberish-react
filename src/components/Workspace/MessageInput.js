import React from 'react';
import PropTypes from 'prop-types';
import { AppBar } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(() => ({
  appBar: {
    top: 'auto',
    bottom: 0,
    marginLeft: '220px',
    width: 'calc(100% - 220px)',
  },
}));

const MessageInput = ({ onSubmitMessage, onUpdateMessageInput, messageInput }) => {

  const classes = useStyles();

  return (
    <AppBar position="fixed" color="inherit" className={classes.appBar}>
      <form onSubmit={onSubmitMessage}>
        <input onChange={onUpdateMessageInput} value={messageInput} autoComplete="off" />
        <button>Send</button>
      </form>
    </AppBar>
  );
};

MessageInput.propTypes = {
  onSubmitMessage: PropTypes.func.isRequired,
  onUpdateMessageInput: PropTypes.func.isRequired,
  messageInput: PropTypes.string.isRequired
};

export default MessageInput;
