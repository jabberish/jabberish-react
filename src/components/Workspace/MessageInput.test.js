/* eslint-disable no-console */
import React from 'react';
import { shallow } from 'enzyme';
import MessageInput from './MessageInput';

describe('MessageInput component', () => {
  it('renders MessageInput', () => {
    const wrapper = shallow(<MessageInput 
      onSubmitMessage={() => {}} 
      onUpdateMessageInput={() => {}} 
      messageInput='input'
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
