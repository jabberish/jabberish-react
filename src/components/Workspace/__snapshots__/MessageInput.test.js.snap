// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageInput component renders MessageInput 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-container-1"
  maxWidth="xl"
>
  <WithStyles(ForwardRef(Toolbar))>
    <WithStyles(ForwardRef(TextField))
      autoComplete="off"
      className="makeStyles-input-2"
      multiline={true}
      onChange={[Function]}
      placeholder="Message"
      value="input"
    />
    <WithStyles(ForwardRef(Button))
      onClick={[Function]}
    >
      Send
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(Container))>
`;
